es.go

```diff
@@ -901,7 +911,64 @@ func (c *Client) GetShards(nodes []string) ([]Shard, error) {
 
 	if err != nil {
 		fmt.Printf("Error parsing JSON. Expected ']'")
+		return nil, err
 	}
 
 	return response, nil
 }
+
+//Get details regarding shard distribution across a given set of cluster nodes.
+//
+//Use case: You can leverage this information to determine if it's safe to remove cluster nodes without losing data.
+func (c *Client) GetShardOverlap(nodes []string) (map[string]ShardOverlap, error) {
+	shards, err := c.GetShards(nodes)
+	overlap := map[string]ShardOverlap{}
+
+	if err != nil {
+		fmt.Printf("Error getting shards: %s", err)
+		return nil, err
+	}
+
+	_indices, err := c.GetIndices()
+
+	if err != nil {
+		fmt.Printf("Error getting indices: %s", err)
+		return nil, err
+	}
+
+	// Map-ify this slice of indices for easy lookup
+	indices := map[string]Index{}
+	for _, index := range _indices {
+		indices[index.Name] = index
+	}
+
+	for _, shard := range shards {
+		// Map key is the concatenation of the index name and shard number
+		name := shard.Index + shard.Shard
+
+		val, ok := overlap[name]
+		if ok {
+			// We've already seen this index/shard combo
+			if shard.Type == "p" {
+				val.PrimaryFound = true
+			} else {
```

Comment:
This logic isn't true all of the time. Shards can be in an "initializing" state or even "unassigned", in that case they shouldn't be counted as a replica.

Reply:
To clarify, the `Type` will still be `p` for primary or `r` for replica, but unless the `State` is `STARTED` then it's not a complete replica.

Reply:
I've added a couple of private functions to the ShardOverlap that only considers the shard as primary or replica if the state is "STARTED" or "RELOCATING"